magento/templates/deployment.yaml

```diff
@@ -0,0 +1,83 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "fullname" . }}
+  labels:
+    app: {{ template "fullname" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: {{ template "fullname" . }}
+        chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+        release: "{{ .Release.Name }}"
+        heritage: "{{ .Release.Service }}"
+    spec:
+      containers:
+      - name: {{ template "fullname" . }}
+        image: "bitnami/magento:{{ .Values.imageTag }}"
+        imagePullPolicy: {{ default "" .Values.imagePullPolicy | quote }}
+        env:
+        - name: MARIADB_HOST
+          value: {{ template "mariadb.fullname" . }}
+        - name: MARIADB_PORT
+          value: "3306"
+        - name: MARIADB_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: {{ template "mariadb.fullname" . }}
+              key: mariadb-root-password
+        - name: MAGENTO_HOST
+          value: {{ default .Values.magentoLoadBalancerIP .Values.magentoHost | quote }}
```

Comment:
> Does Magento work if we do not add magentoLoadBalancerIP ? I mean, is it usable?

My tests reveal that specifying `magentoLoadBalancerIP` does indeed result in a working deployment (needed to make some unrelated changes though).

> If it is not the case we should think of making it mandatory.

For certain apps (magento, owncloud, ghost, etc.) it's mandatory to specify the FQDN or external IP address of the service (a.k.a `loadBalancerIP` in case of GKE).

The `magentoLoadBalancerIP` is a way to provide an IP address to associate to the service. This feature is cloud/platform specific and hence cannot be called mandatory IMO. For other clouds users would need to configure and provide the correct hostname via the `magentoHost` parameter at the time of deployment.